kroxylicious/src/test/java/io/kroxylicious/proxy/internal/ResponseOrdererTest.java

```diff
@@ -0,0 +1,212 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.internal;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import io.netty.channel.embedded.EmbeddedChannel;
+
+import io.kroxylicious.proxy.frame.ByteBufAccessor;
+import io.kroxylicious.proxy.frame.Frame;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+class ResponseOrdererTest {
+
+    private EmbeddedChannel embeddedChannel;
+    private ResponseOrderer orderer;
+
+    record TestFrame(int correlationId) implements Frame {
+
+    @Override
+    public int estimateEncodedSize() {
+        return 0;
+    }
+
+    @Override
+    public void encode(ByteBufAccessor out) {
+        throw new IllegalStateException("Not implemented");
+    }
+
+    @Override
+    public int correlationId() {
+        return correlationId;
+    }
+
+    }
+
+    @BeforeEach
+    public void beforeEach() {
+        embeddedChannel = new EmbeddedChannel();
+        orderer = new ResponseOrderer();
+        embeddedChannel.pipeline().addFirst(orderer);
+    }
+
+    @Test
+    void testInboundFrameForwarded() {
+        TestFrame request = new TestFrame(1);
+        whenWriteInboundMessage(request);
+        thenInboundContains(request);
+        thenOrdererStateEquals(1, 0);
```

Comment:
I appreciate this would lead to a bunch of duplication, but I wonder if we should just inline `thenOrdererStateEquals` or make it two separate checks e.g.
```suggestion
        thenThereAreRequestInFlight(1);
        thenThereareQueuedResponses(0);
```

As I don't really feel that `thenOrdererStateEquals` tells a reader whats going on.